test/fn-api-tests/utils.go

```diff
@@ -71,31 +97,43 @@ func getServerWithCancel() (*server.Server, context.CancelFunc) {
 
 		s = server.New(ctx, server.WithDBURL(dbURL), server.WithMQURL(mqURL), server.WithFullAgent())
 
-		go s.Start(ctx)
-		started := false
-		time.AfterFunc(time.Second*10, func() {
-			if !started {
-				panic("Failed to start server.")
-			}
-		})
-		_, err := http.Get(apiURL + "/version")
-		for err != nil {
-			_, err = http.Get(apiURL + "/version")
-		}
-		started = true
-		cancel2 = context.CancelFunc(func() {
-			cancel()
+		go func() {
+			s.Start(ctx)
```

Comment:
should we cancel this context as well if we timeout in the following checkServer loop (I don't see it, but that seems the intention here)?

Reply:
nvm, this seems to be from the enclosing context (not super clear only from patch, my bad)